### PR TITLE
Mat 6526 ucum annotations text field

### PR DESCRIPTION
**Pull request by gregory-akins**
## MADiE PR

Jira Ticket: [MAT-6526](https://jira.cms.gov/browse/MAT-6526)
(Optional) Related Tickets:

### Summary

Replacing drop down with UCUM values with a text field that validates for a Correct "annotation-capable" UCUM value

### All Submissions
* [X] This PR has the JIRA linked.
* [ ] Required tests are included
* [X] No extemporaneous files are included (i.e Complied files or testing results)
* [X] This PR is into the **correct branch**.
* [X] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -128,6 +128,7 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-dropzone": "^14.2.3",
+    "react-icons": "^4.12.0",
```
Comment (nmorasb) [inline package.json]: What was this added for? Is there an icon in this package that we don't get in the regular material UI icons pack that we're using elsewhere?

File: src/components/editTestCase/qdm/QDMPatientSchemaValidator.tsx [hunk 2]
```diff
@@ -1,7 +1,45 @@
 import * as Yup from "yup";
 import { PopulationType } from "@madie/madie-models";
+import * as ucum from "@lhncbc/ucum-lhc";
 
 export const QDMPatientSchemaValidator = Yup.object().shape({
+  // json: Yup.object().test("test-compare a few values", function (value) {
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/QDMPatientSchemaValidator.tsx]: please remove commented code

Comment (sb-prateekkeerthi) [inline src/components/editTestCase/qdm/QDMPatientSchemaValidator.tsx]: all the schema here is commented, are we validating anything here ?

File: src/components/common/quantityInput/QuantityInput.tsx [hunk 3]
```diff
@@ -60,9 +58,19 @@ const QuantityInput = ({
     }
   }, [ucum, ucumUnits]);
 
+  const valueFn = (value: any) => {
+    if (value?.code) {
+      return value.code;
+    } else {
```
Comment (sb-prateekkeerthi) [inline src/components/common/quantityInput/QuantityInput.tsx:65]: Suggestion: we can remove else here and can directly be written as 

 if (value?.code) {
      return value.code;
    } 
    return value

File: src/components/common/quantityInput/QuantityInput.tsx [hunk 4]
```diff
@@ -145,28 +171,20 @@ const QuantityInput = ({
...
-              handleQuantityUnitChange(transformedResult);
-            } else {
-              handleQuantityUnitChange("");
-            }
+          onChange={(e: any) => {
+            handleQuantityUnitChange(e.target.value);
           }}
-          value={currentUnit?.label}
+          value={(() => {
+            return unitText;
```
Comment (sb-prateekkeerthi) [inline src/components/common/quantityInput/QuantityInput.tsx]: why are we writing this in function return format instead of value={unitText}

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (gregory-akins) [reply package.json]: Wasn't necessary

**Review by sb-prateekkeerthi — APPROVED** [review]

**Review by nmorasb — APPROVED** [review]

**Pull request merged**